pkg/server/computeRecommendation.go

```diff
@@ -62,41 +67,51 @@ func computeRecommendations(w http.ResponseWriter, r *http.Request) {
 		}
 		return
 	}
-
+	clusterName := newRecommendation[0].ClusterName
+	nameSpace := newRecommendation[0].Namespace
+	appName := newRecommendation[0].Application
 	//get the clusterID (cluster name with namespace or applicaiton):
-	clusterID := make(map[string]string) //ex: clustername-namespace:"id-12345"
-	var concat string
-
 	//use namespace:
-	if newRecommendation[0].Application == "" && newRecommendation[0].Namespace != "" {
-		concat = fmt.Sprintf("%s-%s", newRecommendation[0].ClusterName, newRecommendation[0].Namespace)
+	if newRecommendation[0].Application == "" && nameSpace != "" {
+		requestName = fmt.Sprintf("ns_%s_%s", clusterName, nameSpace)
 	}
-	//use application
+	//use application , not supported on Dev Preview
 	if newRecommendation[0].Application != "" && newRecommendation[0].Namespace == "" {
-		concat = fmt.Sprintf("%s-%s", newRecommendation[0].ClusterName, newRecommendation[0].Application)
-
+		requestName = fmt.Sprintf("app_%s_%s", clusterName, appName)
 		// if both applications and namespace is empty return
 	} else if newRecommendation[0].Application == "" && newRecommendation[0].Namespace == "" {
 		klog.V(4).Info("Request missing both Application and Namespace. Need at least one to fulfill request.")
 		http.Error(w, "{\"message\":\"Both Application and Namespace cannot be empty.\"}", http.StatusBadRequest)
 		return
 	}
 
-	//if id for clusterName already exists then we don't need to generate new oneß
-	if clusterID[concat] == "" {
-		clusterID[concat] = helpers.GenerateID(clusterID)
+	//if id for clusterName already exists then we don't need to generate new one
+	if _, found := requestIdMap[requestName]; !found {
```

Comment:
What happens when multiple requests come in for the same namespace?

Reply:
We have implemented happy path in this PR. We will have to handle duplicate requests and need to decide the user behavior